chrome/content/zotero/xpcom/openurl.js

```diff
@@ -203,25 +203,33 @@ Zotero.OpenURL = new function() {
 			}
 		}
 		
-		if(item.creators && item.creators.length) {
-			// encode first author as first and last
-			var firstCreator = item.creators[0];
-			if(item.itemType == "patent") {
-				_mapTag(firstCreator.firstName, "invfirst");
-				_mapTag(firstCreator.lastName, "invlast");
-			} else {
-				if(firstCreator.isInstitution) {
-					_mapTag(firstCreator.lastName, "aucorp");
-				} else {
-					_mapTag(firstCreator.firstName, "aufirst");
-					_mapTag(firstCreator.lastName, "aulast");
+		var first = true;
+		for(var i=0; i<item.creators.length; i++) {
+			var creator = item.creators[i];
+			// only export authors
+			// other creator types cannot correctly export to COinS
+			if(creator.creatorType == "author") {
+				// encode first author as aufirst and aulast
+				if (first) {
+					if(item.itemType == "patent") {
+						_mapTag(creator.firstName, "invfirst");
+						_mapTag(creator.lastName, "invlast");
+					} else {
+						_mapTag(creator.firstName, "aufirst");
+						_mapTag(creator.lastName, "aulast");
+					}
```

Comment:
Currently, institutional authors are outputed as `au` and the first one as `aulast`, e.g.
```
rft.aulast=Zotero
rft.au=Zotero
rft.au=Mendeley%20Inc.
```
However, after correcting the `isInstitutional` to `fieldMode` the behaviour would be
```
rft.aucorp=Zotero
rft.au=Zotero
rft.au=Mendeley%20Inc.
```
Is this what we wanted? Sorry, this is really confusing and in the documentation I haven't yet found such an example. At least I have read in the specification that `aucorp` can occur at most once. And somehow this is in line with all the other fields which are only present for the first author. I will change accordingly.